hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestSparkSqlWithCustomKeyGenerator.scala

```diff
@@ -44,451 +42,407 @@ class TestSparkSqlWithCustomKeyGenerator extends HoodieSparkSqlTestBase {
-                Seq(3, "a3", "30.0", 1706800227, "cat1"),
-                Seq(4, "a4", "113.4", 1701443427, "cat2"),
-                Seq(5, "a5", "2009.0", 1704121827, "cat2"),
-                Seq(6, "a6", "99.0", 1704121827, "cat3"),
-                Seq(7, "a7", "1409.0", 1706800227, "cat1"),
-                Seq(8, "a8", "24.9", 1706800227, "cat3"),
-                Seq(9, "a9", "299.0", 1701443427, "cat4"),
-                Seq(10, "a10", "888.8", 1706800227, "cat5")
-              )
-
-              // Test SQL DELETE
-              spark.sql(
-                s"""
-                   | DELETE FROM $tableName
-                   | WHERE id = 7
-                   | """.stripMargin)
-              validateResults(
-                tableName,
-                sqlStr,
-                extractPartition,
-                tsGenFunc,
-                partitionGenFunc,
-                Seq(),
-                Seq(2, "a2", "11.9", 1704121827, "cat1"),
-                Seq(3, "a3", "30.0", 1706800227, "cat1"),
-                Seq(4, "a4", "113.4", 1701443427, "cat2"),
-                Seq(5, "a5", "2009.0", 1704121827, "cat2"),
-                Seq(6, "a6", "99.0", 1704121827, "cat3"),
-                Seq(8, "a8", "24.9", 1706800227, "cat3"),
-                Seq(9, "a9", "299.0", 1701443427, "cat4"),
-                Seq(10, "a10", "888.8", 1706800227, "cat5")
-              )
-
-              // Test DROP PARTITION
-              assertTrue(getSortedTablePartitions(tableName).contains(droppedPartition))
-              spark.sql(
-                s"""
-                   | ALTER TABLE $tableName DROP PARTITION $dropPartitionStatement
-                   |""".stripMargin)
-              validatePartitions(tableName, Seq(droppedPartition), expectedPartitions)
-
-              if (HoodieSparkUtils.isSpark3) {
-                // Test INSERT OVERWRITE, only supported in Spark 3.x
-                spark.sql(
-                  s"""
-                     | INSERT OVERWRITE $tableName
-                     | SELECT 100 as id, 'a100' as name, 299.0 as price, 1706800227 as ts, 'cat10' as segment
-                     | """.stripMargin)
-                validateResults(
-                  tableName,
-                  sqlStr,
-                  extractPartition,
-                  tsGenFunc,
-                  partitionGenFunc,
-                  Seq(),
-                  Seq(100, "a100", "299.0", 1706800227, "cat10")
-                )
-              }
-
-              // Validate ts field is still of type int in the table
-              validateTsFieldSchema(tablePath, "ts", Schema.Type.INT)
-              if (useOlderPartitionFieldFormat) {
-                val metaClient = createMetaClient(spark, tablePath)
-                assertEquals(metaClient.getTableConfig.getPartitionFieldProp, HoodieTableConfig.getPartitionFieldPropForKeyGenerator(metaClient.getTableConfig).orElse(""))
-              }
-            }
+    withTempDir { tmp =>
```

Comment:
These changes are much more than what's in #11710.  Could you strictly revert what's changed?